scripts/metric_reporter/averages_reporter.py

```diff
@@ -0,0 +1,174 @@
+        }
+
+
+class AveragesReporter(BaseReporter):
+    """Handles the reporting of test suite results from CircleCI metadata and JUnit XML Reports."""
+
+    def __init__(
+        self,
+        repository: str,
+        workflow: str,
+        test_suite: str,
+        suite_results: Sequence[SuiteReporterResult],
+    ) -> None:
+        """Initialize the reporter with the test suite results.
+
+        Args:
+            repository (str): The repository associated to the test suite.
+            workflow (str): The workflow associated to the test suite.
+            test_suite (str): The test suite name.
+            suite_results (Sequence[SuiteReporterResult]): Test suite results.
+        """
+        super().__init__()
+        self.results = self._parse_results(repository, workflow, test_suite, suite_results)
+
+    def _parse_results(
+        self,
+        repository: str,
+        workflow: str,
+        test_suite: str,
+        suite_results: Sequence[SuiteReporterResult],
+    ) -> list[AveragesReporterResult]:
+        results: list[AveragesReporterResult] = []
+        if not suite_results:
+            return results
+
+        # Assumes that suite_results is sorted
+        first_date: str | None = suite_results[0].date
+        last_date: str | None = suite_results[-1].date
+        if not first_date or not last_date:
+            return results
+
+        first_datetime: datetime = datetime.strptime(first_date, DATE_FORMAT)
+        last_datetime: datetime = datetime.strptime(last_date, DATE_FORMAT)
+
+        while last_datetime >= first_datetime + timedelta(days=DAYS_30):
+            averages_dict = {
+                "repository": repository,
+                "workflow": workflow,
+                "test_suite": test_suite,
+                **self._calculate_averages(suite_results, last_datetime, DAYS_30),
+                **self._calculate_averages(suite_results, last_datetime, DAYS_60),
+                **self._calculate_averages(suite_results, last_datetime, DAYS_90),
+            }
+            results.append(AveragesReporterResult(**averages_dict))
+            last_datetime -= timedelta(days=DAYS_1)
+        results.reverse()  # Reverse the list to get chronological order
+        return results
+
+    @staticmethod
+    def _calculate_averages(
+        results: Sequence[SuiteReporterResult], stop_date: datetime, delta: int
+    ) -> dict[str, Any]:
+        start_date: datetime = stop_date - timedelta(days=delta)
+        subset: list[SuiteReporterResult] = [
+            r
+            for r in results
+            if r.date and start_date <= datetime.strptime(r.date, DATE_FORMAT) <= stop_date
+        ]
+        suite_count: int = len(subset)
+        averages = {
+            f"start_date_{str(delta)}": start_date.strftime(DATE_FORMAT),
+            f"stop_date_{str(delta)}": stop_date.strftime(DATE_FORMAT),
+            f"suite_count_{str(delta)}": suite_count,
+        }
+
+        if suite_count > 0:
```

Comment:
If this isn't true we still return `averages` which would be a dict with empty values for the strings above. Would this cause any issues?

Reply:
It should not, and this is a possible scenario.
We have a test for it in `test_averages_reporter.py` titled `test_averages_reporter_output_csv_with_empty_test_results`